include/cpr/callback.h

```diff
@@ -12,44 +12,61 @@ class ReadCallback {
   public:
     ReadCallback() = default;
     // NOLINTNEXTLINE(google-explicit-constructor, hicpp-explicit-conversions)
-    ReadCallback(std::function<bool(char* buffer, size_t& size)> p_callback)
-            : size{-1}, callback{std::move(p_callback)} {}
-    ReadCallback(cpr_off_t p_size, std::function<bool(char* buffer, size_t& size)> p_callback)
-            : size{p_size}, callback{std::move(p_callback)} {}
+    ReadCallback(std::function<bool(intptr_t userdata, char* buffer, size_t& size)> p_callback, intptr_t p_userdata = 0)
+            : userdata(p_userdata), size{-1}, callback{std::move(p_callback)} {}
+    ReadCallback(cpr_off_t p_size, std::function<bool(intptr_t userdata, char* buffer, size_t& size)> p_callback, intptr_t p_userdata = 0)
+            : userdata(p_userdata), size{p_size}, callback{std::move(p_callback)} {}
+    bool operator()(char* buffer, size_t& size) const {
+        return callback(userdata, buffer, size);
+    }
 
+    intptr_t userdata;
     cpr_off_t size{};
-    std::function<bool(char* buffer, size_t& size)> callback;
+    std::function<bool(intptr_t userdata, char* buffer, size_t& size)> callback;
```

Comment:
Since `userdata` is an optional argument, we should move it to the last position everywhere to keep it consistent.

Reply:
The userdata parameter, which must appear in the callback function, so, I don't think it's a problem right.

Reply:
Yes, but please change the order, so `userdata` is always the last parameter.

Reply:
All right